### PR TITLE
Improve password reset email copy

### PR DESCRIPTION
**Pull request by saliehendricks**
Fix for #1978 

Simple fix to update the password reset email copy:
image

### CODE REVIEW

**Comment by britttttk** [comment]
ack 

@saliehendricks looks good. I'm not sure about the copy and paste link though. It looks kind of long and intimidating to non-tech people. Is that typical? @brainharrington @Eskyee any feedback? :)

**Comment by Eskyee** [comment]
@saliehendricks looks good.  but I agree with @britttttk 
the copy & paste does looks intimidating to newbies & messy yes.



BTCPay Logo

Please confirm your password.  by clicking the button or link below. 

Button





I think this is the standard Forgot My Password reply. it might be nice just to have a single button 
"if we have a button?? I would remove the link word" and BTCPay Logo in header  included in forgotten email..  
would look more professional and clean.  I think!!

**Comment by PresidentNick** [comment]
The previous string was setting the email copy on whether the user was creating a new password or resetting their existing password.

@saliehendricks Can you add a conditional string for when the user is setting their password for the first time, please? 

Although, I haven't seen what calls the email function with the new password boolean set to True. So maybe it doesn't make sense to have an extra string for when the user is setting a new password? When is this function called with True for newPassword argument?

**Comment by saliehendricks** [comment]
@britttttk I have seen fallback urls in authentication emails, but i dont disagree its pretty techy and ugly tbh. Happy to remove that part.

@Eskyee Agree, a button would be nice. Will try to put something together. I wasn't even sure the email was html formatted, but it appears to be. BTCPayServer logo would be pretty cool. Might as well give it a try :)

@PresidentNick The initial issue that was set out did say it should just be 1 email regardless of it being for a new user. Lets see what I can do to smooth out the copy a little - maybe we can get it done with just 1 simple sentence and no conditionals.

**Comment by saliehendricks** [comment]
I think we making progress. So far this is what I have:
image

Still need to confirm location/file/css regarding the styling & logo with some other devs before committing.

**Comment by Eskyee** [comment]
That's sexi 💯 
Have a look around here for any design tips https://design.btcpayserver.org
Good job 👏

**Comment by Eskyee** [comment]
@dennisreimann could fast track that 🙌

**Comment by dennisreimann** [comment]
Thanks for the pointer @Eskyee, I'll review this tomorrow :)

**Comment by saliehendricks** [comment]
Thanks @dennisreimann. 

My thinking is to create /email.css (currently I just have static strings which is not great). This requires plumbing in the schema:host:port in order to link to the css+images. Let me know if that sounds ok.

**Comment by dennisreimann** [comment]
I would not get too fancy here, especially since styling email content is a pain. Most clients don't load external content and if we want to go with HTML emails we should

- inline css and images, so I think having a general email layout file the content gets wrapped with would be best
- keep it as streamlined as possible
- provide a plain text version too

**Comment by britttttk** [comment]
I like the button @saliehendricks 💯, good idea @Eskyee

File: BTCPayServer/Extensions/EmailSenderExtensions.cs [hunk 1]
```diff
@@ -14,8 +14,8 @@ public static void SendEmailConfirmation(this IEmailSender emailSender, string e
         public static void SendSetPasswordConfirmation(this IEmailSender emailSender, string email, string link, bool newPassword)
         {
             emailSender.SendEmail(email,
-                $"{(newPassword ? "Set" : "Reset")} Password",
-                $"Please {(newPassword ? "set" : "reset")} your password by clicking here: <a href='{link}'>link</a>");
+                $"Update Password",
```
Comment (Kukks) [inline BTCPayServer/Extensions/EmailSenderExtensions.cs:17]: I don't think this should be a static "Update" password. If the user was created by an admin and no pass is set, it's more a "set" password as it's a creation action
```suggestion
                $"{(newPassword ? "Set" : "Update")}  Password",
```

**Review by Kukks — CHANGES_REQUESTED** [review]
I suggest we stick to just the copy text improvement for the scope of this PR. We can do more advanced layout email changes separately to keep things moving

File: BTCPayServer/Extensions/EmailSenderExtensions.cs (hunk 1 above)
Comment (saliehendricks) [reply BTCPayServer/Extensions/EmailSenderExtensions.cs:17]: @Kukks @dennisreimann Cool cool, will keep it to just a copy update. Will adjust the subject as above.

A new feature request can be opened for html emails.

Comment (saliehendricks) [reply BTCPayServer/Extensions/EmailSenderExtensions.cs]: @Kukks @dennisreimann I thought before I strip out the html parts we could review and discuss what we have so far. Kept it as simple as possible (no logo) just inline styles for a button and html font.

image

Comment (dennisreimann) [reply BTCPayServer/Extensions/EmailSenderExtensions.cs]: As this is explicitely setting the font color – have you checked this works in dark mode too? (E.g. for user's that are using dark mode via system preferences)

Comment (saliehendricks) [reply BTCPayServer/Extensions/EmailSenderExtensions.cs]: @dennisreimann I don't think dark mode affects emails?

**Review by NicolasDorier — APPROVED** [review]

**Comment by NicolasDorier** [comment]
fine for me, ping @Kukks , if that's ok for you, you can merge.

**Review by Kukks — APPROVED** [review]
Looks good to me now. We can iterate with nicer HTML stuff later.

Good job @saliehendricks 👍

**Pull request merged**